### PR TITLE
Dataset meta file for some formats

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

Added meta file support in following formats:
- CamVid
- COCO
- KITTI Raw
- LabelMe
- Mapillary Vistas
- MOT
- MOTS
- Open Images
- VggFace2

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/plugins/labelme_format.py [hunk 1]
```diff
@@ -53,9 +54,15 @@ def __init__(self, path):
...
         items = []
         subsets = set()
-        categories = { AnnotationType.label:
-            LabelCategories(attributes={ 'occluded', 'username' })
-        }
+
+        if has_meta_file(dataset_root):
+            categories = { AnnotationType.label:
+                LabelCategories(attributes={ 'occluded', 'username' }).
+                from_iterable(parse_meta_file(dataset_root).keys()) }
```
Comment (IRDonch) [inline datumaro/plugins/labelme_format.py:61]:
```suggestion
                    from_iterable(parse_meta_file(dataset_root).keys()) }
```

File: datumaro/plugins/mot_format.py [hunk 2]
```diff
@@ -264,8 +267,11 @@ def apply(self):
                     else:
                         log.debug("Item '%s' has no image", item.id)
 
-        labels_file = osp.join(anno_dir, MotPath.LABELS_FILE)
-        with open(labels_file, 'w', encoding='utf-8') as f:
-            f.write('\n'.join(l.name
-                for l in extractor.categories()[AnnotationType.label])
-            )
+        if self._save_dataset_meta:
+           self._save_meta_file(self._save_dir)
```
Comment (IRDonch) [inline datumaro/plugins/mot_format.py:271]:
```suggestion
            self._save_meta_file(self._save_dir)
```

File: datumaro/plugins/vgg_face2_format.py [hunk 3]
```diff
@@ -64,8 +65,13 @@ def categories(self):
...
         label_cat = LabelCategories()
-        path = osp.join(self._dataset_dir, VggFace2Path.LABELS_FILE)
-        if osp.isfile(path):
+        if has_meta_file(self._dataset_dir):
+            labels = parse_meta_file(self._dataset_dir).keys()
+            for label in labels:
+                label_cat.add(label)
+        elif osp.isfile(osp.join(self._dataset_dir,
+                VggFace2Path.LABELS_FILE)):
+            path = osp.join(self._dataset_dir, VggFace2Path.LABELS_FILE)
```
Comment (IRDonch) [inline datumaro/plugins/vgg_face2_format.py:74]: Duplicate `osp.join` call.

File: datumaro/plugins/vgg_face2_format.py [hunk 4]
```diff
@@ -199,15 +205,18 @@ def _get_name_id(item_parts, label_name):
...
-        labels_file = ''
-        for label in self._extractor.categories()[AnnotationType.label]:
-            labels_file += '%s' % label.name
-            if label.parent:
-                labels_file += ' %s' % label.parent
-            labels_file += '\n'
-        with open(labels_path, 'w', encoding='utf-8') as f:
-            f.write(labels_file)
+        if self._save_dataset_meta:
+           self._save_meta_file(save_dir)
```
Comment (IRDonch) [inline datumaro/plugins/vgg_face2_format.py:209]:
```suggestion
            self._save_meta_file(save_dir)
```

**Review by IRDonch — CHANGES_REQUESTED** [review]

**Comment by yasakova-anastasia** [comment]
@IRDonch, I fixed this points.

**Review by IRDonch — APPROVED** [review]

**Pull request merged**